modules/claims_api/lib/bd/bd.rb

```diff
@@ -22,7 +22,15 @@ def search(claim_id, file_number)
       body = { data: { claimId: claim_id, fileNumber: file_number } }
       ClaimsApi::Logger.log('benefits_documents',
                             detail: "calling benefits documents search for claimId #{claim_id}")
-      client.post('documents/search', body)&.body&.deep_symbolize_keys
+      res = client.post('documents/search', body)&.body
+
+      raise ::Common::Exceptions::GatewayTimeout.new(detail: "Gateway timeout. #{res}") unless res.is_a?(Hash)
+
+      res.deep_symbolize_keys
+    rescue ::Common::Exceptions::GatewayTimeout => e
+      ClaimsApi::Logger.log('benefits_documents',
```

Comment:
I'd personally put this in the `unless` above, but eh, personal preference

Reply:
I considered logging in the `unless` block too, but then we would just have:

```rb
rescue ::Common::Exceptions::Gatewaytimeout =>
  raise
```

here. At least with the log statement here, it seems like the `rescue` block is doing _something_ before re-raising the exception 🤷 but happy to switch since it works either way.

Reply:
Though it may be better to log in the `unless` if we want to capture `res`. 🤔

Reply:
(aside: we may get the logging for free if we decide to [treat all exceptions the same way](https://github.com/department-of-veterans-affairs/vets-api/pull/17389#discussion_r1664537960)).